pump/src/main/java/org/mvndaemon/mvnd/pump/PumpThread.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mvndaemon.mvnd.pump;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.io.UncheckedIOException;
+
+public class PumpThread extends Thread {
+
+    private final InputStream stream;
+    private final PrintStream out;
+
+    public static void start(InputStream stream, PrintStream out) {
+        new PumpThread(stream, out).start();
+    }
+
+    public PumpThread(InputStream stream, PrintStream out) {
+        this.stream = stream;
+        this.out = out;
+    }
+
+    @Override
+    public void run() {
+        try {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+            try {
```

Comment:
A try(with) would work with less lines of code, wouldn't it?

Reply:
Well, at the beginning, it was more like:
```
new Thread(() -> new BufferedReader(new InputStreamReader(stream)).lines().forEach(out::println));
```
but then I tried to use javaassist to generate this code and came back to using an additional jar.  I haven't reverted to the original one liner code, but I can.